File: Pelikula.DAO/Model/Korisnik.cs
using System;
using System.Collections.Generic;

// Code scaffolded by EF Core assumes nullable reference types (NRTs) are not used or disabled.
// If you have enabled NRTs for your project, then un-comment the following line:
// #nullable disable

namespace Pelikula.DAO.Model
{
    public partial class Korisnik
    {
        public Korisnik()
        {
            Anketa = new HashSet<Anketa>();
            AnketaOdgovorKorisnik = new HashSet<AnketaOdgovorKorisnik>();
            Dojam = new HashSet<Dojam>();
            Obavijest = new HashSet<Obavijest>();
            Prodaja = new HashSet<Prodaja>();
            ProjekcijaKorisnik = new HashSet<ProjekcijaKorisnik>();
            Rezervacija = new HashSet<Rezervacija>();
        }

        public int Id { get; set; }
        public int TipKorisnikaId { get; set; }
        public string KorisnickoIme { get; set; }
        public string LozinkaHash { get; set; }
        public string LozinkaSalt { get; set; }
        public string Ime { get; set; }
        public string Prezime { get; set; }
        public string Email { get; set; }
        public string Spol { get; set; }
        public DateTime? DatumRodjenja { get; set; }
        public byte[] Slika { get; set; }
        public byte[] SlikaThumb { get; set; }

        public virtual TipKorisnika TipKorisnika { get; set; }
        public virtual ICollection<Anketa> Anketa { get; set; }
        public virtual ICollection<AnketaOdgovorKorisnik> AnketaOdgovorKorisnik { get; set; }
        public virtual ICollection<Dojam> Dojam { get; set; }
        public virtual ICollection<Obavijest> Obavijest { get; set; }
        public virtual ICollection<Prodaja> Prodaja { get; set; }
        public virtual ICollection<ProjekcijaKorisnik> ProjekcijaKorisnik { get; set; }
        public virtual ICollection<Rezervacija> Rezervacija { get; set; }
    }
}

File: Pelikula.MOBILE/lib/model/korisnik.dart
import 'dart:convert';

import 'tip_korisnika.dart';

class KorisnikResponse {
  int? id;
  String? korisnickoIme;
  String? ime;
  String? prezime;
  String? email;
  String? spol;
  DateTime? datumRodjenja;
  List<int>? slika;
  List<int>? slikaThumb;
  String? lozinkaHash;
  String? lozinkaSalt;
  String? lozinka;

  TipKorisnikaResponse? tipKorisnika;

  KorisnikResponse(
      {this.id,
      this.korisnickoIme,
      this.ime,
      this.prezime,
      this.email,
      this.spol,
      this.datumRodjenja,
      this.slika,
      this.slikaThumb,
      this.lozinkaHash,
      this.lozinkaSalt,
      this.lozinka,
      this.tipKorisnika});

  factory KorisnikResponse.fromJson(Map<String, dynamic> json) {
    return KorisnikResponse(
        id: json['id'] as int,
        korisnickoIme: json['korisnickoIme'] as String,
        ime: json['ime'] as String,
        prezime: json['prezime'] as String,
        email: json['prezime'] as String,
        spol: json['spol'] as String,
        datumRodjenja: DateTime.tryParse(json['datumRodjenja']),
        slika: json['slika'] != null
            ? base64.decode(json['slika'] as String)
            : null,
        slikaThumb: json['slikaThumb'] != null
            ? base64.decode(json['slikaThumb'] as String)
            : null,
        lozinkaHash: json['lozinkaHash'] as String,
        lozinkaSalt: json['lozinkaSalt'] as String,
        lozinka: json['lozinka'] != null ? json['lozinka'] as String : null,
        tipKorisnika: TipKorisnikaResponse.fromJson(json['tipKorisnika']));
  }

  Map<String, dynamic> toJson() => {
        "id": id,
        "korisnickoIme": korisnickoIme,
        "ime": ime,
        "prezime": prezime,
        "email": email,
        "spol": spol,
        "datumRodjenja":
            datumRodjenja == null ? null : datumRodjenja!.toIso8601String(),
        "slika": slika,
        "slikaThumb": slikaThumb,
        "lozinkaHash": lozinkaHash,
        "lozinkaSalt": lozinkaSalt,
        "lozinka": lozinka,
        "tipKorisnika": tipKorisnika!.toJson(),
      };
}

class KorisnikRegistracijaRequest {
  String? korisnickoIme;
  String? ime;
  String? prezime;
  String? email;
  String? spol;
  DateTime? datumRodjenja;
  String? lozinka;

  KorisnikRegistracijaRequest(
      {this.korisnickoIme,
      this.ime,
      this.prezime,
      this.email,
      this.spol,
      this.datumRodjenja,
      this.lozinka});

  factory KorisnikRegistracijaRequest.fromJson(Map<String, dynamic> json) {
    return KorisnikRegistracijaRequest(
        korisnickoIme: json['korisnickoIme'] as String,
        ime: json['ime'] as String,
        prezime: json['prezime'] as String,
        email: json['prezime'] as String,
        spol: json['spol'] as String,
        datumRodjenja: DateTime.tryParse(json['datumRodjenja']),
        lozinka: json['lozinka'] as String);
  }

  Map<String, dynamic> toJson() => {
        "korisnickoIme": korisnickoIme,
        "ime": ime,
        "prezime": prezime,
        "email": email,
        "spol": spol,
        "datumRodjenja":
            datumRodjenja == null ? null : datumRodjenja!.toIso8601String(),
        "lozinka": lozinka,
      };
}
